modules/core/src/main/java/org/apache/synapse/api/API.java

```diff
@@ -257,7 +276,7 @@ boolean canProcess(MessageContext synCtx) {
                 return false;
             }
         } else {
-            String path = RESTUtils.getFullRequestPath(synCtx);
+            String path = ApiUtils.getFullRequestPath(synCtx);
             if (null == synCtx.getProperty(RESTConstants.IS_PROMETHEUS_ENGAGED) &&
                     (!RESTUtils.matchApiPath(path, context))) {
```

Comment:
matchApiPath is deprecated, please update this one. Also please check none of the deprecated ones are used in synapse itself or anywhere in wso2 code

Reply:
Addressed in d24336b